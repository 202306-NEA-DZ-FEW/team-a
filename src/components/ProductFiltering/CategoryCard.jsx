import Image from "next/image";
import Link from "next/link";

function CategoryCard({ imageUrl, title, link, queryParams }) {
  return (
    <div
      className='card w-60 h-40 card-compact border-none relative rounded-3x'
      aria-label={title}
    >
      <Link
        scroll={false}
        key={link}
        href={{
          pathname: "/products",
          query: { ...queryParams, category: link },
        }}
      >
        <figure className='w-56 h-40 relative rounded-3xl shadow-xl'>
          <Image
            src={imageUrl}
            alt={title}
            className='object-cover hover:scale-125 transition-all duration-700'
            fill
            sizes='(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw'
            priority
          />
          <h3 className='text-white text-center absolute bottom-[10%] px-2 py-1 w-[90%] text-md card-bordered rounded-xl bg-black bg-opacity-30'>
            {title}
          </h3>
        </figure>
      </Link>
    </div>
  );
}

export default CategoryCard;
