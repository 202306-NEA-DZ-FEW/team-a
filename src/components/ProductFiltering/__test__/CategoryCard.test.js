import image from "public/images/mepi1.png";
import renderer from "react-test-renderer";

import CategoryCard from "../CategoryCard";

test("Navbar snapshot test", () => {
  const title = "title";
  const link = "link";
  const queryParams = {};
  const tree = renderer
    .create(
      <CategoryCard
        key={0}
        imageUrl={image}
        title={title}
        link={link}
        queryParams={queryParams}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
