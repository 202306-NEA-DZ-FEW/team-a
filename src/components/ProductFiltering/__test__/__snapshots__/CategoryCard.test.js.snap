// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar snapshot test 1`] = `
<div
  aria-label="title"
  className="card w-60 h-40 card-compact border-none relative rounded-3x"
>
  <a
    href="/products?category=link"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    <figure
      className="w-56 h-40 relative rounded-3xl shadow-xl"
    >
      <img
        alt="title"
        className="object-cover hover:scale-125 transition-all duration-700"
        data-nimg="fill"
        decoding="async"
        fetchpriority="high"
        onError={[Function]}
        onLoad={[Function]}
        sizes="100vw"
        src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
        srcSet="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
        style={
          Object {
            "bottom": 0,
            "color": "transparent",
            "height": "100%",
            "left": 0,
            "objectFit": undefined,
            "objectPosition": undefined,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": "100%",
          }
        }
      />
      <h3
        className="text-white text-center absolute bottom-[10%] px-2 py-1 w-[90%] text-md card-bordered rounded-xl bg-black bg-opacity-30"
      >
        title
      </h3>
    </figure>
  </a>
</div>
`;
